src/backend/columnar/cstore_metadata_tables.c

```diff
@@ -613,7 +613,13 @@ GetHighestUsedAddress(RelFileNode relfilenode)
 {
 	uint64 highestUsedAddress = 0;
 	uint64 highestUsedId = 0;
-	ColumnarMetapage *metapage = ReadMetapage(relfilenode, false);
+	ColumnarMetapage *metapage = ReadMetapage(relfilenode, true);
+
+	/* empty data file? */
+	if (metapage == NULL)
+	{
+		return 0;
```

Comment:
can we safely return 0 as the lowest address here? eg. can this ever be called to place a stripe without having the meta page initialized?

Reply:
I updated the PR description. This is not called for stripe placement, but is called in the truncate stage of VACUUM, which is possible even if the data file is empty.

The common function `GetHighestUsedAddressAndId()` is called for stripe placement.